In [3]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

Q1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [4]:
from google.colab import files
uploaded=files.upload()

Saving Cutlets.csv to Cutlets.csv


In [5]:
df=pd.read_csv("Cutlets.csv")

In [6]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [8]:
unit_A=df.iloc[:,0]
Unit_B=df.iloc[:,1]

In [11]:
stats_value,p_value=stats.ttest_ind(unit_A,Unit_B)
print(stats_value,p_value)

0.7228688704678063 0.4722394724599501


Ans) As p-value(0.4722) is greater than alpha significance(0.05) we can conclude that - "We fail to reject null hypothesis i.e. their is no difference in the diameters of the cutlets."

**Q2) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.**

In [12]:
from google.colab import files
uploaded=files.upload()

Saving LabTAT.csv to LabTAT.csv


In [13]:
df_2=pd.read_csv("LabTAT.csv")

In [14]:
df_2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [15]:
lab1=df_2.iloc[:,0]
lab2=df_2.iloc[:,1]
lab3=df_2.iloc[:,2]
lab4=df_2.iloc[:,3]

In [16]:
stat_value,p_value = stats.f_oneway(lab1,lab2,lab3,lab4)
print(stat_value,p_value)

118.70421654401437 2.1156708949992414e-57


Ans) As p-value(2.1156) is greater than alpha significance(0.05) we can conclude that - "We accept alternate hypothesis i.e. their is no significant inter-relation between all the laboratories"

.**Q3) Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.t**

In [17]:
from google.colab import files
uploaded=files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


In [18]:
df_3=pd.read_csv("BuyerRatio.csv")

In [19]:
df_3.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [20]:
Males = df_3.iloc[0,1:]
Females = df_3.iloc[1,1:]

In [21]:
data1=[Males,Females]
print(data1)

[East      50
West     142
North    131
South     70
Name: 0, dtype: object, East      435
West     1523
North    1356
South     750
Name: 1, dtype: object]


In [22]:
result = stats.chi2_contingency(data1)
print(result)

(1.595945538661058, 0.6603094907091882, 3, array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))


In [23]:
stat_value = result[0]
pi_value = result[1]
df = result[2]
stat_value, pi_value, df

(1.595945538661058, 0.6603094907091882, 3)

In [25]:
expected_value = result[3]
expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

Ans) As p-value(0.6603) is greater than alpha significance(0.05) we can conclude that - "We fail to reject null hypothesis i.e. All proportions are equal."

**Q4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences**

In [26]:
from google.colab import files
uploaded=files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [29]:
df_4=pd.read_csv("Costomer+OrderForm.csv")

In [30]:
df_4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [31]:
phillippines = df_4.Phillippines.value_counts()
indonesia = df_4.Indonesia.value_counts()
malta = df_4.Malta.value_counts()
india = df_4.India.value_counts()

In [32]:
phillippines.values
indonesia.values
malta.values
india.values

array([280,  20])

In [33]:
df_5 = [phillippines[0],indonesia[0],malta[0],india[0]]
df_6 = [phillippines[1],indonesia[1],malta[1],india[1]]
df = [df_5,df_6]
df

[[271, 267, 269, 280], [29, 33, 31, 20]]

In [34]:
result = stats.chi2_contingency(df)
result

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [35]:
stat_value = result[0]
pi_value = result[1]
df = result[2]
stat_value, pi_value, df

(3.858960685820355, 0.2771020991233135, 3)

Ans) As p-value(0.2771) is greater than alpha significance(0.05) we can conclude that - "We fail to reject null hypothesis i.e. All proportions are equal."